### ISSUE TITLE
Extend reader rights to API importer role

### ISSUE THREAD

**Issue by pisces-period** [DefectDojo/django-DefectDojo#4384]
**Is your feature request related to a problem? Please describe**

The new `api_importer` role allows an account, scoped within a product or product type, to import/re-import scans. According to the Object Based Authorization Model, by design the API importer is not able to see (read) the engagement(s), test(s) or product(s) to which it is associated. Its purpose is strictly to import or re-import a scan.

This presents a problem, however. When I assign the `api_importer` role to an account to interact with DefectDojo via API in an automated way, it is not possible to determine whether it should import or re-import a scan (e.g., by evaluating if a test ID already exists for a particular engagement and what is the scan date of the original scan). It is quite limiting.

This situation has been briefly  discussed [here](https://github.com/DefectDojo/django-DefectDojo/issues/4347).

**Describe the solution you'd like**

As a DevSecOps engineer, I want to use the `api_importer` role to programatically import/re-import scans into DefectDojo via API without having to grant unnecessary privileges, such as create/delete products etc., but that still leaves room for more sophisticated programmatic access to scan related data, such as scan dates, engagement and test IDs etc.

This can be achieved by extending the `reader` rights to the `api_importer` role - thus we'd be able to programatically evaluate and apply control flows for scripted interactions with the DefectDojo API.

**Describe alternatives you've considered**
The alternative solution is to use other roles such as `writer`, `owner` or `maintainer`, which grant more privileges than required.

While this approach works, I wouldn't want to distribute tokens for automated interactions with the API which grant more privileges than needed to import/re-import scans.

**Comment by StefanFl** [comment]
I think this idea makes sense.

**Comment by StefanFl** [comment]
What do the maintainers think about it? @valentijnscholten , @damiencarol, @madchap, @Maffooch

**Comment by damiencarol** [comment]
I will face the same situation in my own company/use case @StefanFl . it make sense I think. but @pisces-period could we just give the 2 roles to your technical account? (`reader` and `api_importer`)

**TLDR; I agree**

**Comment by valentijnscholten** [comment]
I was hoping we could assign multiple roles to a user, but if that's not possible, then yes api import should have read permission to most things related to products/engagements/tests.
Depending on the usecase, it may need more permissions. In my CI/CD jobs I am creating engagements when needed for example. Might even want to create Products when needed. So pretty soon you end up at a writer role.
